features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Resources/components/operationComponents/ListParameters.jsx

```diff
@@ -64,106 +67,158 @@ export default function ListParameters(props) {
+                        <TableCell align='left'>
+                            <FormattedMessage
+                                id='Apis.Details.Resources.components.operationComponents.ListParameter.parameter.type'
+                                defaultMessage='Parameter Type'
+                            />
+                        </TableCell>
+                        <TableCell>
+                            <FormattedMessage
+                                id='Apis.Details.Resources.components.operationComponents.ListParameter.parameter.name'
+                                defaultMessage='Name'
+                            />
+                        </TableCell>
+                        <TableCell align='left'>
+                            <FormattedMessage
+                                id='Apis.Details.Resources.components.operationComponents.ListParameter.data.type'
+                                defaultMessage='Data Type'
+                            />
+                        </TableCell>
+                        <TableCell align='left'>
+                            <FormattedMessage
+                                id='Apis.Details.Resources.components.operationComponents.ListParameter.required'
+                                defaultMessage='Required'
+                            />
+                        </TableCell>
+                        {!disableUpdate && (
+                            <TableCell align='left'>
+                                <FormattedMessage
+                                    id='Apis.Details.Resources.components.operationComponents.ListParameter.actions'
+                                    defaultMessage='Actions'
+                                />
+                            </TableCell>
+                        )}
                     </TableRow>
                 </TableHead>
                 <TableBody>
                     {operation.parameters
-                        && operation.parameters.map((parameter) => (
-                            <TableRow key={parameter.name}>
-                                <TableCell align='left'>{parameter.name}</TableCell>
-                                <TableCell align='left'>{capitalizeFirstLetter(parameter.in)}</TableCell>
-                                <TableCell align='left'>
-                                    {capitalizeFirstLetter(parameter.schema ? parameter.schema.type : parameter.type)}
-                                </TableCell>
-                                <TableCell align='left'>{parameter.required ? 'Yes' : 'No'}</TableCell>
-                                {!disableUpdate && (
+                        && operation.parameters.map((parameter, index) => {
+                            const isRefParam = isRef(parameter);
+                            const paramCopy = isRefParam
+                                ? resolvedSpec.paths[target][verb].parameters[index] : parameter;
```

Comment:
Can we rely on index here , will it be always the same in the raw definition param array index and parsed definition params array index ? what if we find by param name ?

Reply:
Some times, the whole parameter object can be put as a ref. There we cannot find any information.

parameters:
- $ref: '#/definitions/query'
- $ref: '#/definitions/path'